### ISSUE TITLE
chore: add workshop to teach typescript to frontend cert

### ISSUE THREAD

**Issue by jdwilkin4** [freeCodeCamp/CurriculumExpansion#556]
We will have a workshop to teach typescript fundamentals to campers.
You can see what videos will come before this on the sheet.

This will be a mid sized project. Maybe 40-50 steps?

we will supply the css for them. So there will be no steps having them write that out.

### Acceptance criteria

- [ ]  add your workshop files to the react project folder in the front-end cert directory

**Comment by jdwilkin4** [comment]
marking this as blocked for now until we know what the final editor implementation will look like for the react and typescript projects.
Oliver will let us know when that is

**Comment by jdwilkin4** [comment]
The TypeScript changes have now been merged into main, so we can start working on thi

**Comment by jdwilkin4** [comment]
This will resume in 2025 Q1

**Comment by a2937** [comment]
For the workshop, I'm pretty sure that we should introduce first introduce the concept of typing in Typescript in variables and functions. Then get campers familiar with union types and maybe mention the pitfalls. I think introducing inheritance and interfaces afterwards might be a good idea. 

I'm fairly sure I forgot to mention something but this should be the gist.

**Comment by jdwilkin4** [comment]
@a2937 

There will be a series of lecture videos that goes before this workshop.
But it might be better to break this workshop up into smaller workshops instead of one mid sized one.
I will bring it up in the curriculum meeting and see what we decide from there 👍

**Comment by jdwilkin4** [comment]
we are going to break this up into small workshops and have Naomi handle that

**Comment by tekluabayneh** [comment]
butterfly

**Comment by naomi-lgbt** [comment]
Thanks Teklu!

**Comment by shootermv** [comment]
can it look simple like this ui?

<img width="617" height="472" alt="Image" src="https://github.com/user-attachments/assets/c404824a-ee35-4448-891e-6f017a322348" />
it has only one button "Reveal Fortune"
when user presses it - it shows a random text from fortunes array

https://github.com/user-attachments/assets/bae40107-f1a5-48fd-93b6-b8b923b05fe7

**Comment by jdwilkin4** [comment]
Yeah simple is fine. This will be for the main curriculum and most of those designs are pretty simple. 
We just have to make sure any images are free to use

**Comment by tekluabayneh** [comment]
but we use api for it right?

**Comment by shootermv** [comment]
which typescript topics it better to contain?

**Comment by jdwilkin4** [comment]
@tekluabayneh @shootermv 

You will need to coordinate with naomi on which topics this workshop is supposed to focus on because she did the initial planning for this.
